### ISSUE TITLE
Suggested Course Outline?

### ISSUE THREAD

**Issue by kmoon3** [mml-book/mml-book.github.io#506]
I received a physical copy of the book from Cambridge Press and so far it looks very nice. Great work! It was also quite lucky as I am planning on developing a course at my university this Fall titled "Mathematical Methods for Data Science" with a focus on multivariate calculus and linear algebra. 

My question is, do you have a suggested outline for the book when teaching it as a course? Do you recommend covering the whole book in a single semester? Or is that typically too much material? I'm interested in hearing about your experience in using the materials for any courses you've taught.

For the planned audience, I will be teaching it as an upper division undergraduate and lower division graduate course in a Math and Statistics department.  However, we are also targeting students from other departments who have had at least one calculus course and some exposure to linear algebra.

Thanks in advance!

**Comment by chengsoonong** [comment]
The following is a rough outline for a course at the Australian National University. Third year computer science course, run over 12 weeks, with 3 hours of lectures (2 x 1.5 hours), and 2 hour practical session (computer laboratory/tutorial) per week.

It uses material beyond the book (k-means and logistic regression). Since it is a computer science class, it tries to alternate between mathematical content and algorithms.

| Week | Topic | Notes |
| ------ | ------ | ------ |
| 1 | Introduction | administration, high level |
| 1 | Linear Regression (univariate) | Section 8.1, motivating need for linear algebra |
| 2 | Linear Algebra | Chapter 2 | 
| 2 | Analytic Geometry | Chapter 3 |
| 3 | Analytic Geometry | Chapter 3 |
| 3 | Model Meets Data | Chapter 8, not formal |
| 4 | Model Meets Data | Chapter 8, not formal |
| 4 | Clustering with k-means | not in book, use to illustrate Chapter 8. |
| 5 | Vector calculus | Chapter 5 |
| 5 | Vector calculus + gradient descent | Chapter 5, Section 7.1 |
| 6 | Logistic regression (binary) | not in book, use to illustrate need for gradients | 
| 6 | Logistic regression | motivate ideas of probability |
| 7 | Probability and distribution | Chapter 6 |
| 7 | Probability and distribution | Chapter 6 |
| 8 | Gaussian mixture models | Chapter 11 |
| 8 | Gaussian mixture models | Chapter 11 |
| 9 | Matrix decompositions | Chapter 4 |
| 9 | Principal Component Analysis | Chapter 10 |
| 10 | Model Meets Data | Chapter 8, formal, MLE, ERM |
| 10 | Model Meets Data | Chapter 8, formal, motivate need for optimization from loss functions |
| 11 | Continuous Optimization | Section 7.2 |
| 11 | Continuous Optimization | Section 7.3 |
| 12 | Support Vector Machine | Chapter 12 |
| 12 | Support Vector Machine | Chapter 12 |
